src/main/java/com/gruzini/tennistico/services/MatchService.java

```diff
@@ -29,13 +31,19 @@ public Match save(final Match match) {
         return matchRepository.save(match);
     }
 
-    public void updateExpiredMatchesStatus(final LocalDateTime expirationDateTime, final MatchStatus currentStatus, final MatchStatus newStatus) {
-        final List<Match> matchesToUpdate = getAllExpiredByStatus(expirationDateTime, currentStatus);
-        updateMatchStatus(matchesToUpdate, newStatus);
+    public void updateExpiredMatchesStatus(final ChangeMatchStatusEvent event) {
+        List<Match> matchesToUpdate = getAllExpiredByStatus(event.getExpirationDateTime(), event.getCurrentMatchStatus(), event.getDateTimeSelector());
+        updateMatchStatus(matchesToUpdate, event.getDesiredMatchStatus());
     }
 
-    private List<Match> getAllExpiredByStatus(final LocalDateTime expirationDateTime, final MatchStatus matchStatus) {
-        return matchRepository.findByStartingAtBeforeAndMatchStatus(expirationDateTime, matchStatus);
+    private List<Match> getAllExpiredByStatus(final LocalDateTime expirationDateTime,
+                                              final MatchStatus matchStatus,
+                                              final DateTimeSelector dateTimeSelector) {
+        if(dateTimeSelector.equals(DateTimeSelector.STARTING_AT)){
+            return matchRepository.findByStartingAtBeforeAndMatchStatus(expirationDateTime, matchStatus);
+        } else {
+            return matchRepository.findByEndingAtBeforeAndMatchStatus(expirationDateTime, matchStatus);
+        }
```

Comment:
Chyba lepiej niż if, dać tutaj dwie metody i robić if'a w Listenerze jak już. Tak myślę. Do dyskusji.

Reply:
miałem chyba tak nawet na początku, ale to spowodowało, że w serwisie i tak byś miał 2 metody, jedyne o czym pomyślałem w drodze do sklepu, że mogę zmienić DateTimeSelector po prostu na boola, przynajmniej nie będzie tworzony dodatkowy enum